File: week2/AhmetTahaSener/LongestSubstringWithoutRepeatingCharacters.cs
public class Solution
{
    public int LengthOfLongestSubstring(string s)
    {
        HashSet<char> substr = new HashSet<char>();
        int l = 0, r = 0; // left, right index
        int longestSubstring = 0;

        while (r < s.Length)
        {
            if (!substr.Contains(s[r]))
            {
                substr.Add(s[r++]);
                longestSubstring = Math.Max(longestSubstring, r - l);
            }
            else
            {
                substr.Remove(s[l++]);
            }
        }

        return longestSubstring;
    }
}

File: week2/Erenimo3442/longest_substring_without_repeating_characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        unique = set()
        answer = 0
        for right in range(len(s)):
            if s[right] not in unique:
                unique.add(s[right])
                answer = max(answer, right - left + 1)
            else:
                while s[right] in unique:
                    unique.remove(s[left])
                    left += 1
                unique.add(s[right])
        return answer